src/unix/udp.c

```diff
@@ -533,6 +603,17 @@ int uv__udp_bind(uv_udp_t* handle,
 #endif
   }
 
+#if defined(__linux__)
```

Comment:
This `
#if defined(__linux__)` check should be inside `if (flags & UV_UDP_PKTINFO)` and return UV_ENOTSUP in case of non linux systems

Reply:
We won't ever enable this feature in non-linux system, so I'll not make the change for now.